File: OOP/3_private_encaptulaiton.cpp
#include<bits/stdc++.h>
using namespace std;

class Car{
    //private varable can't access outsite the class
    private:
    string name;
    string engine;
    int price;

    public:
    //constructor
    Car(string name, string engine,int price){
        this->name = name;
        this->engine = engine;
        this->price=price;
    }

    //method
    void display(){
        cout<<"Car is "<< name <<" and engine " <<engine <<" price " << price <<endl;
    }

};


int main(){
  Car car("BMW","MOTOR",3232);
  car.display();
  // cout<<car.name;  // declared private here
              
    return 0;
}

File: OOP/3_private_encaptulaiton.py
class Car:
    campany_name = "Volvo"  #class attributes
    def __init__(self,name,engine,price) -> None:
        #instance atributes
        #name engine public attribues and price private attributes  can't access  outside the class
        self.name=name
        self.engine=engine
        self.__price=price
    
    def display(self):
        print(f'Company name {self.campany_name} car {self.name} price {self.__price}')
    
    def car_price(self):
        print(f"Car price {car.__price}")

    def __str__(self) -> str:
        return f'Company name {self.campany_name}  '

car=Car('BMW',"Motor",2312)
car.display()
car.car_price()

print(car.name)
# print(car.__price) # can't access  because price private varible



